R/makeTreeSummarizedExperimentFromqiime2.R

```diff
@@ -1,93 +1,99 @@
-#'   qiime2, default `NULL`.
+#' @param sampleMetaFile character, file path of the sample metadata in tsv
+#'  format, default `NULL`.
+#' @param featureNamesAsRepseq logical, only worked while argument `repSeqFile`
+#'   is `NULL`. The default is `TRUE`, which means if the name of feature is a
+#'   DNA sequence, use it as representative sequence of the feature.
+#' @param repSeqFile character, file path of the representative sequence for
+#'   each feature, default `NULL`.
+#' @param phyTreeFile character, file path of the phylogenetic tree, default
+#'   `NULL`.
 #' @return  An object of class
 #'   [`TreeSummarizedExperiment::TreeSummarizedExperiment-class`]
+#' @importFrom Biostrings DNAStringSet
 #' @export
+#' @author Yang Cao
+#' @references \url{https://qiime2.org}
 #' @examples
-#' otuqza_file <- system.file("extdata", "table.qza", package = "mia")
-#' taxaqza_file <- system.file("extdata", "taxonomy.qza", package = "mia")
-#' sample_file <- system.file("extdata", "sample-metadata.tsv", package = "mia")
-#' treeqza_file <- system.file("extdata", "tree.qza", package = "mia")
-#' refseqqza_file <- system.file("extdata", "refseq.qza", package = "mia")
+#' featureTableFile <- system.file("extdata", "table.qza", package = "mia")
+#' taxonomyTableFile <- system.file("extdata", "taxonomy.qza", package = "mia")
+#' sampleMetaFile <- system.file("extdata", "sample-metadata.tsv", package = "mia")
+#' phyTreeFile <- system.file("extdata", "tree.qza", package = "mia")
+#' repSeqFile <- system.file("extdata", "refseq.qza", package = "mia")
 #' tse <- makeTreeSummarizedExperimentFromqiime2(
-#'   otu_qza = otuqza_file, taxa_qza = taxaqza_file,
-#'   sam_tab = sample_file, refseq_qza = refseqqza_file,
-#'   tree_qza = treeqza_file
+#'   featureTableFile = featureTableFile,
+#'   taxonomyTableFile = taxonomyTableFile,
+#'   sampleMetaFile = sampleMetaFile,
+#'   repSeqFile = repSeqFile,
+#'   phyTreeFile = phyTreeFile
 #' )
 #'
 #' tse
-makeTreeSummarizedExperimentFromqiime2 <- function(otu_qza,
-                                                   taxa_qza = NULL,
-                                                   sam_tab = NULL,
-                                                   refseq_qza = NULL,
-                                                   tree_qza = NULL) {
-    feature_tab <- read_qza(otu_qza)
-
-    if (!is.null(taxa_qza)) {
-        taxa_tab <- read_qza(taxa_qza)
-        taxa_tab <- subset_taxa_in_feature(taxa_tab, feature_tab)
-        taxa_tab <- parse_q2taxonomy(taxa_tab)
+makeTreeSummarizedExperimentFromqiime2 <- function(featureTableFile,
+                                                   taxonomyTableFile = NULL,
+                                                   sampleMetaFile = NULL,
+                                                   featureNamesAsRepseq = TRUE,
+                                                   repSeqFile = NULL,
+                                                   phyTreeFile = NULL) {
```

Comment:
Since the accessors to the reference sequences, is called `referenceSeq` and the internal slot is named `refSeq`, I would have expected an argument to be called `refSeqFile`, instead of `repSeqFile`. Care to comment on this decision?

The behavior of `featureNamesAsRepseq` and `repSeqFile` needs to be explained as mentioned in the last review. Please include a details section in the roxygen comment section above.